### PR TITLE
Update form-bindings.md to include info about component auto-recovery

### PR DESCRIPTION
**Pull request by deepankar-j**
See Issue #1355.

### CODE REVIEW

File: guides/client/form-bindings.md [hunk 1]
```diff
@@ -210,6 +210,18 @@ above, which would wire up to the following server callbacks in your LiveView:
...
 To forgo automatic form recovery, set `phx-auto-recover="ignore"`.
 
+### Recovery within components
+
+Form recovery within components is only supported when the component is statically added
+(i.e. it is present after mounting is completed).  Recovery is not supported within dynamically
+added components.  For example, components that may be shown conditionally using an
+`<%= if ... do>` statement.
+
+For dyanmically added components, there are two options for auto-recovery:
```
Comment (richardkmichael) [inline guides/client/form-bindings.md:220]: dynamically

File: guides/client/form-bindings.md [hunk 2]
```diff
@@ -210,6 +210,18 @@ above, which would wire up to the following server callbacks in your LiveView:
...
+
+- Convert the component to be statically added and have the component handle a hidden state
```
Comment (richardkmichael) [inline guides/client/form-bindings.md:222]: "..., and handle visibility with component state" might be a bit clearer about the state being in the component (if that's what you mean) ?

File: guides/client/form-bindings.md [hunk 3]
```diff
@@ -210,6 +210,18 @@ above, which would wire up to the following server callbacks in your LiveView:
...
+- Backup your form values to the database and restore on mount
```
Comment (richardkmichael) [inline guides/client/form-bindings.md:223]: I'd avoid second-person and just refer to the values directly: "Backup the form values", ... "restore them on mount"

**Review by richardkmichael — COMMENTED** [review]
Thanks for adding these docs, this might have surprised me too. :)

**Comment by chrismccord** [comment]
Thanks for holding tight on this! I think the docs are incomplete in this PR since the main issue is we cannot auto restore *state* for you, of which that state changes at some point dynamically added your component, which does not get recovered. So it's not specifically about dynamically added components themselves, it's more about state restoration. If you want to give a second pass, please let us know. Thanks!

**Comment by richardkmichael** [comment]
@chrismccord Thanks for commenting! Sorry, I'm having trouble understanding your sentence: "of which that state changes at some point dynamically added your component".  Could you clarify?

**Comment by dmorneau** [comment]
I think if the form is within an if/else block, the recovery will work as long as you ensure the LV renders the same if/else branch once the page reload. You can do this by storing state in the URL for instance, using `live_patch` / `push_patch` and `handle_params`.

If you generate a new app with Phoenix (at least master branch) that's how modals work: https://github.com/phoenixframework/phoenix/blob/d91f2cb3d89498b67bdcf177caca84c5ba21c98c/priv/templates/phx.gen.live/modal_component.ex

**Pull request closed without merging**